mcs/class/System/System.Security.Cryptography.X509Certificates/X509ChainImplMono.cs

```diff
@@ -580,6 +580,62 @@ private void Process (int n)
 
 			// TODO 6.1.3.f - verify explict_policy > 0 if valid_policy_tree != NULL
 		}
+		
```

Comment:
Please do not put valuable new code into a class that is scheduled for retirement.

Since you have already added a new internal PAL-based class for `X500DistinguishedName`, that would be the perfect place for these two methods and make them public in that internal class.  I definitely see value in having a new managed implementation of this, it just shouldn't be hidden in here but find a home where it can be reused in other code as well.

Reply:
It is not a new code. The code is implementation that was formerly present in X500DistinguishedName, but used only in this one place. I moved it over.

Reply:
Well, it looks like it's general-purpose code that could possibly be reused elsewhere and I also want to retire `X509ChainImplMono` soon.

Reply:
It does look general purpose, but it also looks like the equality is not very well defined - at least for this particular implementation. Right now the method is not used anywhere else and CoreFX implements the chain in the PAL in a different way.

Reply:
Once X509ChainImplMono (& the rest of the legacy code) gets retired then this patch will shrink considerably.